### ISSUE TITLE
Blank lines at the end of a file are now disallowed with 3.12.1

### ISSUE THREAD

**Issue by nebrius** [palantir/tslint#1351]
### Bug Report

- __TSLint version__: 3.12.1
- __TypeScript version__: 1.8.10
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
const foo = 10;

```

note: there is a blank line after `const foo = 10;`, but the GitHub markdown parser strips it from view.

with `tslint.json` configuration:

```json
{
    "rules": {
        "class-name": true,
        "curly": true,
        "eofline": true,
        "forin": true,
        "indent": [ true, "spaces" ],
        "label-position": true,
        "label-undefined": true,
        "jsdoc-format": true,
        "member-access": true,
        "no-any": true,
        "no-arg": true,
        "no-bitwise": true,
        "no-conditional-assignment": true,
        "no-consecutive-blank-lines": true,
        "no-construct": true,
        "no-console": [ true, "debug", "info", "time", "timeEnd", "trace" ],
        "no-debugger": true,
        "no-duplicate-key": true,
        "no-duplicate-variable": true,
        "no-empty": true,
        "no-eval": true,
        "no-internal-module": true,
        "no-null-keyword": true,
        "no-shadowed-variable": true,
        "no-string-literal": true,
        "no-switch-case-fall-through": true,
        "no-trailing-whitespace": true,
        "no-unreachable": true,
        "no-unused-expression": true,
        "no-unused-variable": true,
        "no-use-before-declare": true,
        "no-var-keyword": true,
        "no-var-requires": true,
        "one-line": [ true, "check-open-brace" ],
        "quotemark": [ true, "single", "avoid-escape" ],
        "radix": true,
        "semicolon": true,
        "switch-default": true,
        "trailing-comma": [ true, { "multiline": "never", "singleline": "never" }],
        "triple-equals": [ true, "allow-null-check" ],
        "typedef-whitespace": [ true, { "call-signature": "nospace", "index-signature": "nospace", "parameter": "nospace", "property-declaration": "nospace", "variable-declaration": "nospace" } ],
        "use-strict": [ true, "check-module" ],
        "variable-name": [ true, "allow-leading-underscore" ],
        "whitespace": [ true, "check-branch", "check-decl", "check-operator", "check-module", "check-separator", "check-type"]
    }
}
```

#### Actual behavior

Prior to v 3.12.x, blank lines at the end of a file were allowed, but post 3.12.x this now creates a `Consecutive blank lines are forbidden` error.

#### Expected behavior

Behavior in 3.11.x should be the same as 3.12.x in this case. Many code formatters/linters/editors/etc prefer a blank line at the end of a file.

**Comment by JKillian** [comment]
@nebrius A single newline at the end of your file shouldn't produce this error (it doesn't for us anyways). I think you have a blank line and then an additional new line after that blank line in your files - essentially there's an extra blank line sitting around.

A single newline is standard for the end of a file, but I've never heard of having a blank line before that newline - it seems redundant. Could you point me to any resources/examples of code where this is typically done?

**Comment by nebrius** [comment]
@JKillian Ahh, I see what happened now. We had an editor config set to hide the last newline, so it wasn't readily apparent there were two newlines, and the previous bug that was fixed in 3.12.0 obscured this was happening. Thanks!
